integrations/extensions/starter-kits/google-custom-search/README.md

```diff
@@ -0,0 +1,153 @@
+- Follow the `basic` step below to get started with search and test it in the Preview chat.
+- Results can be filtered or scoped based on the given parameters with advanced filtered search. Once you are done with basic steps, follow the filtered search section for advanced setup using filter parameters in search api.
+  ![Variable](./assets/variables.png)<br>
+
+
+### Basic
+Once this starter kit is properly installed, you can issue a query to your bot and if there is no other action that you've configured that matched that query then it will generate search results for that query. 
+
+#### Search
+- Create new action and name it `Search`.
+- Click the fX button to add a variable and add new session variable `query_text` and select "Expression" type and then put `input.text` or `input.original_text` as the expression.  The former will employ spelling correction to fix any detected spelling errors before sending the query, which can be helpful but it can also be counterproductive if your documents include specialized terminology that is not in our dictionary (such as product names) so you can use `input.original_text` as the alternative in such cases.
+- Optional: In "Assistant says", put `Searching for: ${query_text}`.
+- In "And then", select "Use an extension", and select the search endpoint and set the `query` parameter to the `query_text` session variable and the `cx` parameter to `cx` session variable you set in pre-requisite step and `num` parameter to `num_of_results` session variable. 
+- In new step, Store the results returned by extension in variable `extension_result`.
+- In next steps, call the `Process Result` action to do further processing on the results stored in previous step.
+- If you are planning for advanced `Filtered Search` action. Then you need to add extra step asking user the type of filter they want to apply on the result. Based on the user response each filter will be selected at runtime.
+  ![Search Action](./assets/search_action.gif)<br>
+
+#### Process Result
+- Create new action and name it `Process Result`
+- change "without conditions" to "with conditions" and "check if `extension_result is not defined`  or `extension_result.body == null` or `extension_result.body.items == null` or `extension_result.body.items.size == 0`.
+- In "Assistant says" provide failure response "Sorry search failed! Please try again". 
+- Add new step for the success response if results are not empty.
+- Change "without conditions" to "with conditions" and "check if `extension_result.success==true` and In "And then" section select "Go to another action" and type "Show Search Results" which we will create in further sections.
+  ![Process Result](./assets/process_result.gif)<br>
+
+#### Show Search Results
+- Create new action and name it `Show Search Results`
+- Add a "New Step" then:
+  - Change without conditions to "with conditions" and check if `${extension_result.body.items}.size==0` then add Assistant says:
```

Comment:
I think this is the one that should check `extension_result.body.items == null` or `extension_result.body.items.size == 0` because either of those would indicate that you got no search results.  I think in practice, what happens is `extension_result.body.items == null` for no search results, but it seems good to check both just in case.

Reply:
I have removed the condition from "Show Search Results" since I have included checks in previous action "Process result"